### ISSUE TITLE
How to actually perform post requests?

### ISSUE THREAD

**Issue by anetschka** [aspnet/Docs#7526]
I am trying to perform integration tests on an Asp.Net Core 2.0 Web Api project. I have cookie authentication in place. Get requests work fine. But I seem to be unable to post.

My controller methods return application/json data and are decorated with HttpPost attributes as needed. The object to be posted is decorated with a [FromBody] attribute. The api works fine in Swagger. 

In the test project, I use a custom web application factory to set up server and client with cookie authentication. However, both client.PostAsync (url, stringcontent) and client.SendAsync(httprequestmessage) do not produce POST requests even when I explicitly set the request method to "POST". All requests are treated as GET and, therefore, fail producing a 405 status code. 

Documentation on how to perform proper integration testing for api projects will be greatly appreciated.

**Comment by guardrex** [comment]
Hi @anetschka ... sorry to hear you're having trouble. I'm surprised to hear that `PostAsync` would produce a GET request.

We can ask @javiercn if he has a tip, but you may need to open this where the engineers will have better visibility on it over in [aspnet/Home issues](https://github.com/aspnet/Home/issues). If you find out that the docs need content that would've helped you, we would like to hear about that, of course. Please post back when you find out what was going wrong.

**Comment by javiercn** [comment]
@anetschka Can you file an issue and provide a minimal repro project on the MVC repository at https://github.com/aspnet/mvc?

**Comment by anetschka** [comment]
Hi @guardrex , @javiercn , thanks for your quick reaction. I actually solved the problem. It turned out that the post request was not able to "survive" throughout authentication. The user has to be logged in before posting ...^^. I guess that this issue can be closed. However, for the documentation of integration tests (which is very good, in general), it might be beneficial to either add material specific to integration tests on Web Api projects or to add more sample code beyond the Razor project (or have I missed something?). There were some recurrent requests for documentation concerning authentication and my question might just be another case in point. Best, a

**Comment by guardrex** [comment]
We didn't discuss the auth part of the sample app explicitly beyond a quick mention in the topic and a quick test that an endpoint is secure.

> The app includes a /SecurePage that can only be accessed by an authenticated user.

https://github.com/aspnet/Docs/blob/master/aspnetcore/test/integration-tests/samples/2.x/IntegrationTestsSample/tests/RazorPagesProject.Tests/IntegrationTests/BasicTests.cs#L47-L64

I agree that we should take the topic+sample to the next level with one or a few authenticated/authorized tests. Let's use this issue to track the work. btw- It will take a little while to get to this. I'm slammed for the next several weeks.

**Comment by Rick-Anderson** [comment]
Thanks for contacting us.
We don’t have the resources to invest in this area, so we are closing the issue. Should your request generate enough  👍 responses, we’ll reconsider.

**Comment by AJMcKane** [comment]
We have also recently hit this issue.  It seems the generated client only wants to send GET requests, we've attempted POST and PUT requests both via the _client.PostAsync and _client.SendAsync (with a Message specifying POST/PUT) and we can see that the request is ultimately being sent as GET.  We've stripped back bindings, auth etc and to no avail, so I'm lead to believe this is an issue with the library. 

I'm surprised this isn't a higher rated issue, and also that there's no documented examples for POST or PUT'ing via WebApplicationFactory.
